sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -13,10 +13,33 @@
 
 #include "pi_level_zero.hpp"
 #include "ur_bindings.hpp"
+#ifdef _WIN32
+#include <Windows.h>
+#include <delayimp.h>
+#endif
 
 // Defined in tracing.cpp
 void enableZeTracing();
 void disableZeTracing();
+#ifdef _WIN32
+FARPROC WINAPI delayHook(unsigned dliNotify, PDelayLoadInfo pdli) {
```

Comment:
I will have to remove this hook function from here since we plan to mirror these changes in UR otherwise we will get multiple definition build error. @kbenzie please review my UR changes(https://github.com/oneapi-src/unified-runtime/pull/1117) and I will update this PR accordingly.  FYI, I have tested both the changes together in this draft PR- https://github.com/intel/llvm/pull/12030

Reply:
LGTM